new example: Asian option pricing using 2D PDE (Pawel Magnuszewski's MSc project draft, to be submitted to arXiv soon)

pawelmagnu opened the pull request.

pawelmagnu committed:
init

pawelmagnu committed:
fixes

pawelmagnu committed:
pylint

pawelmagnu committed:
pylint

slayoo committed:
address pylin hints

slayoo committed:
temporarily limit CI runs to just the single new notebook

slayoo committed:
Fig 1: fix secondary grid (shift by dx/2); Fig 2: align all figs, one colorbar, common color ranges, every second tick

slayoo committed:
nan imshow logic fix

slayoo committed:
int div //

pawelmagnu committed:
removed numba flag; added ref to mc code

pawelmagnu committed:
adressed some issues before merge

pawelmagnu committed:
pylint

pawelmagnu committed:
pylint

slayoo commented on examples/PyMPDATA_examples/Magnuszewski_et_al_2025/monte_carlo.py:
```diff
@@ -0,0 +1,121 @@
+"""
+This code is a Python numba-fied implementation of the Monte Carlo method
+for pricing Asian options taken from
+[Numerical Methods in Finance with C++](ISBN 9781107003712)
```
https://doi.org/10.1017/CBO9781139017404 ?

pawelmagnu committed:
doi for mc code

pawelmagnu committed:
added to example gallery

pawelmagnu committed:
further pylint fixes

pawelmagnu committed:
comments to code and overall notebook logic

pawelmagnu committed:
pylint

slayoo committed:
reinstantiate all yml workflow files

slayoo committed:
code cleanups, advector approximation for quiver

slayoo committed:
uncomment commented workflow code

slayoo committed:
fix notebook code

slayoo committed:
shortening simulation on CI; quantizing colors; introducing comparison with UPWIND; removing duplicate simulation run; code cleanups

slayoo committed:
removing unused import

slayoo committed:
addressing pylin hints

slayoo committed:
CI settings and plot zoom fine tuned

slayoo committed:
further plot beef up (contours, cmap, scales)

pawelmagnu committed:
pylint lambda

slayoo committed:
fix github rendering by using inline_format=png since imshow() is used; fine tune colors, sizes, line styles

slayoo committed:
Merge branch 'asian-arxiv' of github.com:pawelmagnu/PyMPDATA into HEAD

slayoo committed:
fix grid placement; annotate where Fig. 2 is locaten in Fig. 2; cbar tick colors, ...

slayoo committed:
add a set of basic smoke tests for the new notebook (one test done, two drafts to be filled-in with code)

pawelmagnu committed:
moved plots around, twinx twiny

pawelmagnu committed:
precommit

pawelmagnu committed:
fix twiny axis ranges

slayoo committed:
- switch from np.interp to scipy.interpolate.make_interp_spline to properly handle extrapolation
- remove obsolete narrative text in code comments
- fix imshow extents axes origin at (-dx/2, -dy/2)
- contour isoline levels fixed to show zero isoline for terminal condition
- fix Fig2 x2ticks (and move then dx/2 from the grid lines)
- code cleanups
- add (a), (b), (c), (d) labels for plots, remove all but top and bottom tick labels, but keep ticks everywhere

slayoo committed:
seed setting for MC; ax2 funcs fixes; tests completed; common var for option type (call/put); remove MC error-bar code (not used); removing plot-func definition; cleanups

slayoo committed:
pylint & precommit

slayoo committed:
addressing pylint hint

slayoo committed:
remove unused import

slayoo committed:
remove nogil opt (does not apply here), add nopython=True (to avoid warnings in some Numba versions - this caused CI to fail)
